pkg/pillar/types/smarttypes.go

```diff
@@ -1,65 +1,88 @@
 /*
- * Copyright (c) 2020. Zededa, Inc.
+ * Copyright (c) 2022. Zededa, Inc.
  * SPDX-License-Identifier: Apache-2.0
  */
 
 package types
 
-//SmartData holds SMART details of the device
-type SmartData struct {
-	SmartCtl        SmartCtl       `json:"smartctl"`
-	Device          Device         `json:"device"`
-	SmartAttribute  SmartAttribute `json:"ata_smart_attributes"`
-	PowerOnTime     PowerOnTime    `json:"power_on_time"`
-	PowerCycleCount int64          `json:"power_cycle_count"`
-	Temperature     Temperature    `json:"temperature"`
-	RawData         string         `json:"raw_data"`
+//DeviceSmartInfo holds SMART details of the device
+type DeviceSmartInfo struct {
+	Smartctl           cmdSmartctl     `json:"smartctl"`
+	Device             device          `json:"device"`
+	ModelName          string          `json:"model_name"`
+	SerialNumber       string          `json:"serial_number"`
+	Wwn                smWWN           `json:"wwn"`
+	AtaSmartAttributes SmartAttributes `json:"ata_smart_attributes"`
+	Temperature        Temperature     `json:"temperature"`
+	PowerOnTime        PowerOnTime     `json:"power_on_time"`
+	PowerCycleCount    int64           `json:"power_cycle_count"`
 }
 
-//Device holds device details
-type Device struct {
+//DisksList holds list with disks
+type DisksList struct {
+	Smartctl cmdSmartctl `json:"smartctl"`
+	Devices  []device    `json:"devices"`
+}
+
+type errMessages struct {
+	String   string `json:"string"`
+	Severity string `json:"severity"`
+}
+
+//cmdSmartctl query result.
+type cmdSmartctl struct {
+	Argv       []string      `json:"argv"`
+	Messages   []errMessages `json:"messages"`
+	ExitStatus int           `json:"exit_status"`
+}
+
+//device holds device details
+type device struct {
 	Name     string `json:"name"`
 	InfoName string `json:"info_name"`
 	Type     string `json:"type"`
 	Protocol string `json:"protocol"`
 }
 
-//SmartAttribute holds SMART attributes
-type SmartAttribute struct {
-	Revision int64   `json:"revision"`
-	Table    []Table `json:"table"`
-}
-
-//Table schema for each SMART attribute
-type Table struct {
-	ID         int64  `json:"id"`
-	Name       string `json:"name"`
-	Value      int64  `json:"value"`
-	Worst      int64  `json:"worst"`
-	Thresh     int64  `json:"thresh"`
-	WhenFailed string `json:"when_failed"`
-	Flags      Flags  `json:"flags"`
-	Raw        Raw    `json:"raw"`
+type smWWN struct {
+	Naa int   `json:"naa"`
+	Oui int   `json:"oui"`
+	ID  int64 `json:"id"`
 }
 
-//Flags of each SMART attribute
-type Flags struct {
-	Value         int64  `json:"value"`
+type smFlags struct {
+	Value         int    `json:"value"`
 	String        string `json:"string"`
-	PreFailure    bool   `json:"prefailure"`
+	Prefailure    bool   `json:"prefailure"`
```

Comment:
Is the to make the CamelCase have a constent mapping to lowercase_dash (I see the other entries have a constent mapping)

Reply:
Removed this code, comment above.